VitruvioHost/Plugins/Vitruvio/Source/Vitruvio/Private/Util/AttributeConversion.cpp

```diff
@@ -211,23 +209,55 @@ bool IsAttributeBeforeOther(const URuleAttribute& Attribute, const URuleAttribut
 		return true;
 	};
 
-	auto GetFirstDifferentGroupInA = [](const URuleAttribute& A, const URuleAttribute& B) {
-		check(A.Groups.Num() == B.Groups.Num());
-		size_t i = 0;
+	auto GetGlobalGroupOrder = [&GlobalGroupOrderMap](const URuleAttribute& RuleAttribute)
+	{
+		const int* GroupOrderPtr = GlobalGroupOrderMap.Find(FGroupOrderKey(RuleAttribute));
+		return (GroupOrderPtr == nullptr) ? AttributeGroupOrderNone : (*GroupOrderPtr);
+	};
+
+	auto AreAttributeGroupsInOrder = [](const URuleAttribute& A, const URuleAttribute& B)
+	{
+		const size_t GroupSizeA = A.Groups.Num();
+		const size_t GroupSizeB = B.Groups.Num();
 
-		while ((i < A.Groups.Num()) && (A.Groups[i] == B.Groups[i]))
+		for (size_t i = 0; i < std::max(GroupSizeA, GroupSizeB); ++i)
 		{
-			i++;
+			// a descendant of b
+			if (i >= GroupSizeA)
+				return false;
+
+			// b descendant of a
+			if (i >= GroupSizeB)
+				return true;

```

Comment:
Sadly we can't return early because we don't know if group length or an alphabetical difference in the group will decide which is sorted first.
Imagine two attributes a and b with multiple groups but the same group order
attr a groups: A>B>C
attr b groups: A>D>E>F
=> a before b

attr a groups: A>B
attr b groups: A
=> B before A